#### Урок 2. Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

##### 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

**Решение:** $$P_n(X=k) = C_n^k p^k q^{n-k}$$

$$P(k=85) = C_{100}^{85}  {0.8}^{85}  {0.2}^{25}$$

In [9]:
from math import factorial, exp

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [3]:
n = 100
k = 85
p = 0.8
X = combinations(n, k) * (p ** k) * (1 - p) ** (n - k)
X

0.048061793700746355

**Ответ:** 0.048

##### 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?


**Решение:** $$P_m \approx \frac{\lambda^m}{m!}e^{-\lambda}$$


In [11]:
n = 5000
m1 = 0
m2 = 2
p = 0.0004
lambda_ = n * p

print(f'Вероятность, что ни одна лампочка не перегорит = {lambda_**m1 / factorial(m1) * exp(-lambda_)}')
print(f'Вероятность, что перегорят ровно 2 лампочки = {lambda_**m2 / factorial(m2) * exp(-lambda_)}')

Вероятность, что ни одна лампочка не перегорит = 0.1353352832366127
Вероятность, что перегорят ровно 2 лампочки = 0.2706705664732254


**Ответ:** 

Вероятность, что ни одна лампочка не перегорит = 0.1353352832366127

Вероятность, что перегорят ровно 2 лампочки = 0.2706705664732254


##### 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

**Решение:** 

$$P_n(X=k) = C_n^k p^k q^{n-k}$$

$$P(k=85) = C_{144}^{70}  {0.5}^{70}  {0.5}^{30}$$

In [12]:
n = 144
k = 70
p = 0.5
X = combinations(n, k) * (p ** k) * (1 - p) ** (n - k)
X

0.06281178035144776

**Ответ:** 0,063

##### 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. a) Какова вероятность того, что все мячи белые? б) Какова вероятность того, что ровно два мяча белые? в) Какова вероятность того, что хотя бы один мяч белый?

**Решение:**

**a)** Умножаем вероятности достать белый шарик из первого ящика: 7/10 * 6/9.
Из второго: 9/11 * 8/10

In [39]:
7/10 * 6/9 * 9/11 * 8/10

0.3054545454545454

In [19]:
p1 = (7/10) * (6/9)
p1

0.4666666666666666

In [20]:
p2 = 9/11 * 8/10
p2

0.6545454545454545

In [24]:
p_total = p1*p2
print(f'Вероятность того, что все мячи белые = {p_total}')

Вероятность того, что все мячи белые = 0.3054545454545454


**б)** Посчитаем варианты:

In [61]:
# вариант 1: 2 белых шара из первой и 0 белых из второй корзин
a = combinations(10,2)
b = combinations(7,2)
c = b/a
c

0.4666666666666667

In [89]:
7/10*6/9

0.46666666666666656

In [84]:
a1 = combinations(11,2)
b1 = combinations(9,0)
c1 = b1/a1
c1

0.01818181818181818

In [90]:
2/11*1/10

0.01818181818181818

In [64]:
p1 = c * c1
p1

0.008484848484848484

In [105]:
# вариант 2: из первой ноль белых, из второй - 2
a2 = combinations(10,2)
b2 = combinations(3,2)
c2 = b2/a2
c2

0.06666666666666667

In [91]:
3/10*2/9

0.06666666666666667

In [107]:
a3 = combinations(11,2)
b3 = combinations(9,2)
c3 = b3/a3
c3

0.6545454545454545

In [95]:
p2 = c2 * c3
p2

0.04363636363636363

In [115]:
#вариант 3: из каждой корзины по 1 белому
c4 = 7/10*3/9
print(f'Вероятность достать из первой корзины сперва белый, затем чёрный = {c4}')

Вероятность достать белый, затем чёрный = 0.23333333333333328


In [134]:
c7 = 2/11*9/10
print(f'Вероятность достать из второй корзины сперва чёрный, затем белый = {c7}')

Вероятность достать из второй корзины сперва чёрный, затем белый = 0.16363636363636364


In [135]:
c8 = c4 * c7
print(f'Общая вероятность = {c8}')

Общая вероятность = 0.03818181818181817


In [136]:
t = 3/10*7/9
print(f'Вероятность достать из первой корзины сперва чёрный, затем белый = {t}')

Вероятность достать из первой корзины сперва чёрный, затем белый = 0.23333333333333334


In [137]:
t1 = 9/11*2/10
print(f'Вероятность достать из второй корзины сперва белый, затем чёрный ={t1}')

Вероятность достать из второй корзины сперва белый, затем чёрный =0.16363636363636364


In [138]:
t2 = t * t1
print(f'Общая вероятность  = {t2}')

Общая вероятность  = 0.038181818181818185


Т.о. полная группа событий равна:

In [140]:
print(f'Т.о. полная группа событий равна: {p1 + p2 + t2}')

Т.о. полная группа событий равна: 0.0903030303030303


**в):** 

In [110]:
a1 = combinations(10,2)
b1 = combinations(3,2)
c1 = b1/a1
print(f'Вероятность достать два чёрных шара из первой корзины = {c1}')

Вероятность достать два чёрных шара из первой корзины = 0.06666666666666667


In [79]:
a2 = combinations(11,2)
b2 = combinations(2,2)
c2 = b2/a2
print(f'Вероятность достать 2 черных шара из второй корзины = {c2}')

Вероятность достать 2 черных шара из второй корзины = 0.01818181818181818


In [55]:
c_total = c2*c1
c_total

0.0012121212121212121

In [56]:
print(f'Т.о. вероятность достать хотя бы один белого цвета шар = {1 - c_total}')

Т.о. вероятность достать хотя бы один белого цвета шар = 0.9987878787878788


**Ответ:** 

a) Вероятность того, что все мячи белые = 0.3054545454545454

б) Вероятность того, что ровно два мяча белые = 0.0903030303030303

в) Вероятность достать хотя бы один белого цвета шар = 0.9987878787878788